File: harmoni_sehat_backend/src/models/Pharmacist.ts
import { Schema, model, Document, Types } from 'mongoose';

export interface IPharmacist extends Document {
  _id: Types.ObjectId;
  user_id: Types.ObjectId;
  nama: string;
  nomor_sipa: string;
  createdAt: Date;
  updatedAt: Date;
}

const pharmacistSchema = new Schema<IPharmacist>({
  user_id: {
    type: Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    unique: true,
  },
  nama: {
    type: String,
    required: true,
    trim: true,
  },
  nomor_sipa: {
    type: String,
    required: true,
    unique: true,
    trim: true,
  },
},
{
  timestamps: true,
});

const Pharmacist = model<IPharmacist>('Pharmacist', pharmacistSchema);

export default Pharmacist;

File: harmoni_sehat_frontend/lib/shared/models/pharmacist.dart
class Pharmacist {
  final String id;
  final String nama;
  final String nomorSipa;

  Pharmacist({
    required this.id,
    required this.nama,
    required this.nomorSipa,
  });

  Map<String, dynamic> toJson() {
    return {'user_id': id, 'nama': nama, 'nomor_sipa': nomorSipa};
  }
}
